Plugin/CppApi/source/ArcGISArView.cpp

```diff
@@ -321,6 +318,47 @@ DeviceOrientation ArcGISArView::toDeviceOrientation(Qt::ScreenOrientations orien
   }
 }
 
+/*!
+  \internal
+
+  Calculate the origin camera to use in TMCC, by addition of the origin camera (wich contains
+  the initial view of the scene view and calibration) and the location camera (which contains
+  the GPS data).
+  m_originCamera and m_locationCamera can be invalid if not set previously.
+ */
+void ArcGISArView::updateTmccOriginCamera() const
+{
+  if (m_originCamera.isEmpty())
+  {
+    if (m_locationCamera.isEmpty())
+    {
+      // Do nothing.
+    }
+    else
+    {
+      m_tmcc->setOriginCamera(m_locationCamera);
+    }
```

Comment:
Is it important to have the `if (m_locationCamera.isEmpty())` -> `// Do nothing.` logic for readability?

If not you could condense this into:
```c++
    if (!m_locationCamera.isEmpty())
    {
      m_tmcc->setOriginCamera(m_locationCamera);
    }
```